src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -35,13 +35,14 @@ internal partial class RemoteJSRuntime : JSRuntime
 
     public RemoteJSRuntime(
         IOptions<CircuitOptions> circuitOptions,
-        IOptions<HubOptions> hubOptions,
+        IOptions<HubOptions<ComponentHub>> hubOptions,
+        IOptions<HubOptions> globalHubOptions,
         ILogger<RemoteJSRuntime> logger)
     {
         _options = circuitOptions.Value;
-        _maximumIncomingBytes = hubOptions.Value.MaximumReceiveMessageSize is null
-            ? long.MaxValue
-            : hubOptions.Value.MaximumReceiveMessageSize.Value;
+        _maximumIncomingBytes = hubOptions.Value.MaximumReceiveMessageSize ??
```

Comment:
`null` is a valid value. So if a user set `MaximumReceiveMessageSize` in the hub specific options to null then they want that hub to have no limit. As this is written you would ignore the user set null value and use the global option instead which might be 10 bytes for all you know.

Reply:
So is there any case where we would care about the global options here or should we only be considering the `ComponentHub` options? Ie, should I just revert this to the prior logic on lines 42-44 (using ComponentHub options instead of global hub options).

Reply:
Should we straight up ignore global options? For e.g. it's valid to do this no:

```C#
.AddSignalR(options => options.MaxRMS = 100)
.AddBlazorServer(options => options.MaxRMS = null)
```

Reply:
> Should we straight up ignore global options

I think the answer is yes, but I'll let Brennan confirm in case there are other considerations.

Reply:
There are a few different scenarios here:
```
.AddSignalR(options => options.MaxRMS = 100)
.AddBlazorServer(options => options.MaxRMS = null) // use blazor server options
```
```
.AddSignalR(options => options.MaxRMS = 100); // use global options
```
```
.AddBlazorServer(options => options.MaxRMS = null) // use blazor server options
```

As the code is written the first scenario would use 100 when it should be `long.MaxValue`.

Reply:
In the middle case there where we want to use the global options, wouldn't the component hub options default to `null` (and thus we would interpret it as `long.MaxValue`)?

I guess what I'm trying to determine is how to determine if the component options are null because the value wasn't provided and we want to use the global options, or if the component hub option was purposely set to null.

Reply:
On second thought, based on the comment above and 

https://github.com/dotnet/aspnetcore/blob/f346a2793d5acee12f2129050c2fb7cda1ac7eb1/src/SignalR/server/SignalR/src/SignalRDependencyInjectionExtensions.cs#L51

it appears that the `ComponentHub` options roll in the global options. So we should only be looking at the component hub options. 

```
_maximumIncomingBytes = componentHubOptions.Value.MaximumReceiveMessageSize ?? long.MaxValue;
```

Reply:
@BrennanConroy is the above understanding correct (always only look at the component hub options as the global options are "rolled in")?

Reply:
> (always only look at the component hub options as the global options are "rolled in")?

Oh, that might be true, shouldn't be too hard to test

Reply:
Verified, the global options are "rolled into" the component hub options. Thanks for the discussion! 😄